ethcore/src/block.rs

```diff
@@ -465,7 +466,10 @@ impl LockedBlock {
 
 impl Drain for LockedBlock {
 	/// Drop this object and return the underlieing database.
-	fn drain(self) -> Box<JournalDB> { self.block.state.drop().1 }
+	fn drain(mut self) -> StateDB {
+		self.block.state.commit_cache();
```

Comment:
if `commit_cache` is now necessary to be called as part of `LockedBlock`'s general maintenance, what guarantees are there that the rest of the codebase is using it correctly?